build_scripts/build_usd.py

```diff
@@ -863,6 +863,12 @@ def InstallTIFF(context, force, buildArgs):
         PatchFile("CMakeLists.txt", 
                    [("add_subdirectory(tools)", "# add_subdirectory(tools)"),
                     ("add_subdirectory(test)", "# add_subdirectory(test)")])
+
+        if MacOS() or iOS():
+            PatchFile("CMakeLists.txt",
+                      [("option(ld-version-script \"Enable linker version script\" ON)",
+                        "option(ld-version-script \"Enable linker version script\" OFF)")])
+
```

Comment:
- There isn't an iOS() function in this script, perhaps this accidentally got pulled in from changes in your fork?

- Since `ld-version-script` is already a CMake option, can we just specify that instead of patching the CMakeLists.txt? Like:

```
if MacOS():
    # Need to turn off ld-version-script because...
    extraArgs.append("-Dld-version-script=OFF")

extraArgs += buildArgs
RunCMake(context, force, extraArgs)
```

- Could we add documentation about why this is necessary? A link to a bug report or something like that would be great.

Reply:
That suggests another option... The problem is that **d-version-script=ON** invokes a command line option that in the latest version of the linker causes an exit. We wouldn't want the default configuration on Big Sur to be failuire, fixable only with an argument everyone would be forced to append... If I can ascertain the correct toolchain identifier, then a better fix would be to set the linker script on or off based on toolchain version.

Reply:
I updated the title and the description to more exactly describe the problem. An update to the fix itself will follow.

Reply:
Patch updated.